aten/src/ATen/native/Blas.cpp

```diff
@@ -137,6 +140,110 @@ Tensor dot(const Tensor &self, const Tensor &other){
   });
 }
 
+// this is to port LDA_COND from the TH codebase
+// #define LDA_COND(M, N, LDA) ((N) == 1 || (LDA) >= THMax(1, (M)))
+// lda_cond function in this file is not used
+// because it requires `lda > std::max<int64_t>(1L, m)`
+// instead of `>=` as defined in the TH implementation
+constexpr inline bool lda_cond_addr(int64_t m, int64_t n, int64_t lda) {
+  return n == 1 || lda >= std::max<int64_t>(1L, m);
+}
+
+template<typename scalar_t>
+Tensor &addr_impl_cpu(Tensor &result, const Tensor &self,
```

Comment:
What if we wrote addr as a composite op:

beta * input + alpha * torch.linalg.outer(vec1, vec2)

? 

Would performance suffer so badly? It would make the code much more maintainable, and addr is a little-used function.

Reply:
What's interesting is that ger appears to be implemented using addr, though:

https://github.com/pytorch/pytorch/blob/b6b5ebc3458c22512284eda263c360bb04a83e52/aten/src/ATen/native/LinearAlgebra.cpp#L167

I guess I would have expected that ger would be ported and the ger functions (torch.ger and torch.linalg.outer) would use it, and then this function could be written using torch.linalg.outer, as above.

Reply:
The performance impact may refer to the table of https://github.com/pytorch/pytorch/pull/44364#issuecomment-692171950